www/js/metrics/footprintHelper.ts

```diff
@@ -0,0 +1,113 @@
+import { displayErrorMsg, logDebug } from '../plugin/logger';
+import { getCustomFootprint } from './CustomMetricsHelper';
+
+//variables for the highest footprint in the set and if using custom
+let highestFootprint = 0;
+let useCustom = false;
```

Comment:
I remember thinking about doing this, but then was worried about what would happen if we hadn't initialized the custom footprint yet, you're right that we are always going to be custom at this point (either deployer-specified or from the sample custom files). I'm going to reevaluate and try and pull this out, but might end up needing some error handling.

Reply:
This was less complex than I thought it would be. I elected to check that the custom footprint is defined before returning it in 'getFootprint`, that way we can still catch if the custom footprint is broken

Reply:
Wondering now if we should apply the same treatment to the METs ... we still need the dataset, since it's what we pull from for met-equivalents but if the labels are always custom then the mets should also always be custom...

Reply:
Concluded yes, the custom mets are populated when we initialized the custom dataset helper, and if they are not defined for some reason we can still fall back to the standard.